src/service.coffee

```diff
@@ -145,14 +139,25 @@ module.exports = (settings)->
 
     es.random options
 
-  service.get '/:type/:id' , jsonP( (req)->
+  service.get '/:type/:id' , jsonP( (req, res, next) ->
     tf = settings.types[req.params.type]?.documentTransform
 
-    es.fetch(req.params.id,req.params.type).then (body)->
-      if tf? and tf.transform? then tf.transform body._source else body._source
+    es.fetch req.params.id, req.params.type
+      .then (body) ->
+        if tf? and tf.transform? then tf.transform body._source else body._source
+      .catch (error) ->
+        if settings.notFoundHandlers?
+          next()
+        else
+          console.error error
+          error
   )
 
+  if settings.notFoundHandlers?
```

Comment:
I really do not like the name `notFoundHandlers`. It should be something more generic. After all, there are perfectly valid use cases for these hooks, that have nothing to do with error handling at all.
Maybe `postPlugins` (as opposed to `plugins`)? Not sure. Any suggestion appreciated.